addons/hr_expense/tests/test_expenses.py

```diff
@@ -794,6 +793,76 @@ def test_payment_edit_fields(self):
 
         payment.write({'is_move_sent': True})
 
+    def test_corner_case_expense_reported_cannot_be_zero(self):
+        """
+        Test that the expenses are not submitted if the total amount is 0.0 nor able to be edited that way
+        unless unlinking it from the expense sheet.
+        """
+        expense = self.create_expense({'total_amount': 0.0, 'total_amount_currency': 0.0})
+
+        # CASE 1: FORBIDS Trying to submit an expense with a total_amount(_currency) of 0.0
+        with self.assertRaises(UserError):
+            expense.action_submit_expenses()
+
+        # CASE 2: FORBIDS Trying to change the total_amount(_currency) to 0.0 when the expense is linked to a sheet
+        expense.total_amount_currency = 1000
+        expense_sheet = expense._create_sheets_from_expense()
+        with self.assertRaises(UserError):
+            expense.total_amount_currency = 0.0
+        with self.assertRaises(UserError):
+            expense.total_amount = 0.0
+
+        # CASE 3: FORBIDS Trying to change the total_amount(_currency) to 0.0 when the expense sheet is approved
+        expense_sheet.action_approve_expense_sheets()
+        with self.assertRaises(UserError):
+            expense.total_amount_currency = 0.0
+        with self.assertRaises(UserError):
+            expense.total_amount = 0.0
```

Comment:
Does it make sense to even be able to update the amount if the sheet has been approved?

Reply:
Yes, the expense admin has that right (an accountant could fix wrong taxes on the expense)